php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsInFunctionCallsCheck.java

```diff
@@ -0,0 +1,223 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.ArgumentMatcher;
+import org.sonar.php.checks.utils.ArgumentVerifierUnaryFunction;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.php.checks.utils.FunctionArgumentCheck;
+import org.sonar.php.tree.impl.declaration.ClassNamespaceNameTreeImpl;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
+import org.sonar.plugins.php.api.tree.expression.NewExpressionTree;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONArray;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONObject;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.JSONParser;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.ParseException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@Rule(key = "S6437")
+public class HardCodedCredentialsInFunctionCallsCheck extends FunctionArgumentCheck {
+
+  private static final String MESSAGE = "Revoke and change this password, as it is compromised.";
+
+  private static final String JSON_LOCATION = "HardCodedCredentialsSensitiveFunctions.json";
+
+  private static final Map<String, SensitiveMethod> SENSITIVE_FUNCTIONS = JsonSensitiveFunctionsReader.parseSensitiveFunctions();
+
+  private static final Logger LOG = Loggers.get(HardCodedCredentialsInFunctionCallsCheck.class);
+
+  @Override
+  public void visitNewExpression(NewExpressionTree tree) {
+    FunctionCallTree functionCallTree = (FunctionCallTree) tree.expression();
+    ClassNamespaceNameTreeImpl callee = (ClassNamespaceNameTreeImpl) functionCallTree.callee();
+    String fqn = callee.symbol().qualifiedName().toString();
+    String methodName = "__construct";
+    checkForSensitiveMethod(functionCallTree, fqn + "::" + methodName);
+
+    super.visitNewExpression(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    ExpressionTree callee = tree.callee();
+
+    if (callee.is(Kind.OBJECT_MEMBER_ACCESS)) {
+      // Is it even possible?
+      // TODO: Remove after more testing
+      throw new RuntimeException("Delete me after testing!");
+    }
+
+    if (callee.is(Kind.CLASS_MEMBER_ACCESS)) {
+      MemberAccessTree memberAccessTreeCallee = (MemberAccessTree) callee;
+      // TODO: Use Nils method
+      // TODO: How will this work, if method is not static? -> Test me!
+      String fqn = ((ClassNamespaceNameTreeImpl) memberAccessTreeCallee.object()).symbol().qualifiedName().toString();
+
+      String methodName = memberAccessTreeCallee.member().toString();
+      checkForSensitiveMethod(tree, fqn + "::" + methodName);
```

Comment:
Same as above. All implementations of `QualifiedName` have their own `toString`. So when only used in concat you do not need to call it before. Nevertheless, please check, why [MemberQualifiedName::toString](https://github.com/SonarSource/sonar-php/blob/8af44b30460bb53e092d0709345dec8217a5c051/php-frontend/src/main/java/org/sonar/php/tree/symbols/MemberQualifiedName.java#L39) is not used. This should return already the FQN including `::`

Reply:
@nils-werner-sonarsource:
I can't use the method right now. In this case `memberAccessTreeCalle.object().symbol().qualifiedName` is an instance of `SymbolQualifiedName` instead of `MemberQualifiedName`

Reply:
Note: This will be handled in a future ticket